torch/csrc/utils/tensor_types.cpp

```diff
@@ -72,7 +72,7 @@ std::vector<std::pair<Backend, ScalarType>> all_declared_types() {
   // can't easily iterate over enum classes
   std::vector<Backend> backends = { Backend::CPU, Backend::CUDA, Backend::SparseCPU, Backend::SparseCUDA };
   std::vector<ScalarType> scalar_types = { ScalarType::Byte, ScalarType::Char, ScalarType::Double, ScalarType::Float,
-                                           ScalarType::Int, ScalarType::Long, ScalarType::Short, ScalarType::Half};
+                                           ScalarType::Int, ScalarType::Long, ScalarType::Short, ScalarType::Half, ScalarType::Bool};
```

Comment:
are we generating a sparse half type?  IMO for now we should keep this the same as SparseHalf, i.e. not supported.

Reply:
We need this in order to have torch.BoolTensor
I've updated the code down below to skip creation of the sparse bool type.